### PR TITLE
Feature/Add Azure Cognitive speech-to-text functionality

### PR DESCRIPTION
**Pull request by ghondar**
Azure Cognitive Services integration for speech-to-text functionality

- Introduced a new credential class for Azure Cognitive Services.
- Updated speech-to-text processing to support Azure Cognitive Services as a provider.
- Enhanced UI components to include Azure Cognitive Services options and inputs for configuration.
- Added necessary imports and error handling for Azure API requests.
Screenshot 2024-12-16 at 5 19 08 PM

### CODE REVIEW

File: packages/components/src/speechToText.ts [hunk 1]
```diff
@@ -3,12 +3,15 @@ import { getCredentialData } from './utils'
 import { type ClientOptions, OpenAIClient, toFile } from '@langchain/openai'
 import { AssemblyAI } from 'assemblyai'
 import { getFileFromStorage } from './storageUtils'
+import axios from 'axios'
+import FormData from 'form-data'
```
Comment (HenryHengZJ) [inline packages/components/src/speechToText.ts:7]: i think this is probably redundant, I've always just directly use the built in `FormData`, can you try to remove this and see if it works fine?

Comment (ghondar) [reply packages/components/src/speechToText.ts:7]: Sure, but when I remove that import, I get this error:
Screenshot 2024-12-17 at 7 57 07 PM

Comment (ghondar) [reply packages/components/src/speechToText.ts]: fixed

**Review by HenryHengZJ — APPROVED** [review]
thank you so much!

**Pull request merged**